Controls/Cafe/Customer1.c

```diff
@@ -19,11 +25,36 @@ void Customer1_checkout(char** name, int* cost) {
  */
 void Task_Customer_1(void* p_arg) {
 	OS_ERR err;	// Make sure to check for errors and print the error code if not OS_ERR_NONE
-	
+	CPU_TS ts; 
 	// Use these values as parameters for Customer1_checkout
 	char* name;
 	int cost;
+	//CPU_INT16U seconds = 4;//4000ms = 4s DELAY
+	
 	while(1){
-		
+		OSMutexPend(&RegisterOccupied_Mutex, 0, OS_OPT_PEND_BLOCKING, &ts, &err);//pendblocking , pendnonblocking doesnt wait, just goes through codes
+		if(err!=OS_ERR_NONE){
+			//printf("Error from OSMutexPend Customer 1");
+			printf("%x;Error from OSMutexPend Customer 1",err);
+			return;
+		}
+		Customer1_checkout(&name,&cost);
+		OSMutexPost(&RegisterOccupied_Mutex, OS_OPT_POST_NONE, &err);
```

Comment:
Your mutex should ideally protect as small a section as possible. You ideally want to protect only the critical sections of the code, so that you aren't sacrificing the performance boost of parallelism for non-critical sections.

Reply:
My suggestion would be to protect only the shared resource that you are changing, which would mean protecting the revenue variable between lines 16 and 18 of this file.